platforms/core-configuration/model-core/src/main/java/org/gradle/internal/extensibility/DefaultExtraPropertiesExtension.java

```diff
@@ -111,18 +111,18 @@ public void setProperty(String name, @Nullable Object newValue) {
     public Map<String, Object> getProperties() {
```

Comment:
💅 I would add some breadcrumbs along the way to highlight that ImmutableMap isn't an option. E.g. we can change the type of `storage` to `Map<String, @Nullable Object>` and/or leave a comment like we do in some other places.

Reply:
I added a note about the contract. `@Nullable` in type position can only be done in master I guess?

Reply:
> `@Nullable` in type position can only be done in master I guess?

You're right. Eventually, we'll get there with NullAway, I hope.